spring-mobile/src/test/java/org/togglz/spring/mobile/DeviceActivationStrategyTest.java

```diff
@@ -0,0 +1,141 @@
+package org.togglz.spring.mobile;
+
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.springframework.mobile.device.Device;
+import org.togglz.core.Feature;
+import org.togglz.core.repository.FeatureState;
+import org.togglz.servlet.util.HttpServletRequestHolder;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.togglz.spring.mobile.DeviceActivationStrategyTest.MockRequest.requestFrom;
+import static org.togglz.spring.mobile.DeviceActivationStrategyTest.MockRequestAssert.assertThat;
+import static org.togglz.spring.mobile.DeviceActivationStrategy.DESKTOP;
+import static org.togglz.spring.mobile.DeviceActivationStrategy.TABLET;
+import static org.togglz.spring.mobile.DeviceActivationStrategy.MOBILE;
+
+
+/**
+ * Created by achhabra on 10/17/16.
+ */
+public class DeviceActivationStrategyTest {
```

Comment:
@hexmind dont want to extract MockRequest to separate class as it contains logic for mocks specific to this strategy only